src/main/java/fi/aalto/cs/apluscourses/model/SubmissionStatusUpdater.java

```diff
@@ -128,4 +118,21 @@ private void run() {
     }
   }
 
+  private boolean fetchResultsAndNotify() {
+    SubmissionResult submissionResult;
+    try {
+      submissionResult =
+          dataSource.getSubmissionResult(submissionUrl, exercise, authentication);
+      if (submissionResult.getStatus() != SubmissionResult.Status.UNKNOWN) {
+        notifier.notifyAndHide(
+            new FeedbackAvailableNotification(submissionResult, exercise), project);
+        PluginSettings.getInstance().updateMainViewModel(project);
+        return true;
+      }
+    } catch (IOException e) {
+      // Fail silently
```

Comment:
@nikke234 , seriously? 😝

Reply:
It's my favorite thing to do when catching exceptions ;) My thinking here was that since it polls the API repeatedly, there would be no reason to notify the user if one attempt fails, but I also wouldn't be opposed to a notification